ephios/core/templatetags/event_extras.py

```diff
@@ -77,3 +78,28 @@ def event_signup_state_counts(event, user):
                 counter[participation.state] += 1
                 break
     return counter
+
+
+@register.simple_tag(name="eventtype_colors")
+def eventtype_colors(request):
+    html = f"<style nonce='{request.csp_nonce}'>"
+    for eventtype in EventType.objects.all():
```

Comment:
Is this a query we can avoid by caching the results of this function?
This could be as simple as a time based cache for the template html...

Reply:
tell me moer about time based cache

Reply:
*in :robot: voice*: Ok, here is more about time based caches: https://docs.djangoproject.com/en/3.1/topics/cache/#template-fragment-caching